alchemy/test/sentry/client-key.test.ts

```diff
@@ -0,0 +1,145 @@
+import { describe, expect } from "bun:test";
+import { alchemy } from "../../src/alchemy.js";
+import { destroy } from "../../src/destroy.js";
+import { SentryApi } from "../../src/sentry/api.js";
+import { ClientKey } from "../../src/sentry/client-key.js";
+import { Project } from "../../src/sentry/project.js";
+import { Team } from "../../src/sentry/team.js";
+import { BRANCH_PREFIX } from "../util.js";
+// must import this or else alchemy.test won't exist
+import "../../src/test/bun.js";
+
+const api = new SentryApi();
+
+const test = alchemy.test(import.meta);
+
+describe("ClientKey Resource", () => {
+  // Use BRANCH_PREFIX for deterministic, non-colliding resource names
+  const testId = `${BRANCH_PREFIX}-test-key`;
+  const projectId = `${BRANCH_PREFIX}-test-project`;
+  const teamId = `${BRANCH_PREFIX}-test-team`;
+  const organization = process.env.SENTRY_ORG;
+  if (!organization) {
+    throw new Error("SENTRY_ORG environment variable is required");
+  }
+
+  test("create, update, and delete client key", async (scope) => {
+    let key: ClientKey | undefined;
+    let project: Project | undefined;
+    let team: Team | undefined;
+    try {
+      // Create a test team first
+      team = await Team(teamId, {
+        name: `Test Team ${teamId}`,
+        slug: teamId,
+        organization,
+      });
+
+      // Create a test project
+      project = await Project(projectId, {
+        name: `Test Project ${projectId}`,
+        slug: projectId,
+        platform: "node-express",
+        team: team.slug,
+        organization,
+      });
+
+      // Create a test client key
+      key = await ClientKey(testId, {
+        name: `Test Key ${testId}`,
+        project: project.slug,
+        organization,
+      });
+
+      expect(key.id).toBeTruthy();
+      expect(key.name).toEqual(`Test Key ${testId}`);
+
+      // Verify key was created by querying the API directly
+      const getResponse = await api.get(
+        `/projects/${key.organization}/${key.project}/keys/${key.id}/`,
+      );
+      expect(getResponse.status).toEqual(200);
+
+      const responseData = await getResponse.json();
+      expect(responseData.name).toEqual(`Test Key ${testId}`);
+
+      // Update the key
+      key = await ClientKey(testId, {
+        name: `Updated Key ${testId}`,
+        project: project.slug,
+        organization,
+      });
+
+      expect(key.id).toEqual(key.id);
+      expect(key.name).toEqual(`Updated Key ${testId}`);
+
+      // Verify key was updated
+      const getUpdatedResponse = await api.get(
+        `/projects/${key.organization}/${key.project}/keys/${key.id}/`,
+      );
+      const updatedData = await getUpdatedResponse.json();
+      expect(updatedData.name).toEqual(`Updated Key ${testId}`);
+    } catch (err) {
+      // log the error or else it's silently swallowed by destroy errors
+      console.log(err);
+      throw err;
```

Comment:
i don't think this is necessary anymore

Reply:
Sorry, which part?  The `try-finally`?  Or the extra `throw`?

Reply:
Found this here:
https://github.com/ericclemmons/alchemy/blob/262757b05322853051fa8297f2a7adfc94f310b3/alchemy/test/neon/project.test.ts#L113-L120

Reply:
The extra throw. 

Yeah it could be removed from neon too. Just a nit pick, not a huge deal.

Reply:
Resolved via cb9ad904b6b1d701aa042d7a4e8e3919fa3bd5b6

Reply:
Actually, the extra throw should be there.

Otherwise, it counts as a Pass:

```console
290 |         // Check response status directly
291 |         if (!response.ok) {
292 |           throw new Error(`API error: ${response.statusText}`);
                      ^
error: API error: Bad Request
      at /Users/eric/Projects/ericclemmons/alchemy/alchemy/src/vercel/project.ts:292:17

Delete:  "eric/project.test.ts/create, update, and delete project/eric-test-project"
Deleted: "eric/project.test.ts/create, update, and delete project/eric-test-project"
Scope is in error, skipping finalize
✓ Project Resource > create, update, and delete project [361.94ms]

 1 pass
 0 fail
 1 expect() calls
```

Reply:
I'll get this cleaned up later tomorrow!

Reply:
You can remove the entire `catch`

Reply:
Resolved via d9dac80dc49c97900d77c1e6b90044e79f710e7e